### DISCUSSION TITLE
How to test runOnJs executed functions?

### DISCUSSION THREAD

**Discussion by Glazzes** [software-mansion/react-native-reanimated#6377] [Q&A] [answered]
Hi there, I'm currently writing a test suite for my zoom library and this one has a lot of callbacks that need to executed within a runOnJS expression, testing with Jest I've noticed passing the mocks to the functions requiring them will not count as an execution, I'm aware of the async nature of runOnJs function, so I guess they're ran after the test is done.

How could I test something like this? Code like this would fail by reporting zero executions were made.
```js
const pan = Gesture.Pan()
  .withTestID('pan')
  .onEnd(e => runOnJS(mock)(e))

// Actual test
test("do something", () => {
  const onPanEnd = jets.fn(() => {});
  
  // Some component with a pan gesture like the one above
  render(<Component onPanEnd={mock} />)

  fireGestureHandler<PanGesture>(getByGestureTestId('pan'), [
    { translationX: 10, translationY: 10, state: State.BEGAN },
    { state: State.ACTIVE },
    { translationX: 10, translationY: 10, state: State.END },
  ]);

  expect(onPanEnd.mock.calls).toHaveLength(1);
});
```

**Comment by Latropos** [comment, marked as answer]
Hi @Glazzes ! could you test the callbacks with some additional shared Value? These values are accessible from both JS and UI threads so you could use them like that:

```
test("do something", () => {
  const someSharedValue = makeMutable(false)
  const onPanEnd = ()=>{'worklet'; someSharedValue.value = true;}
  
  // ... render component and fire the gestue

  expect(someSharedValue.value).toBeTruthy();
});
```

**Reply by Glazzes** [reply]
Hi there @Latropos , this is pretty cool as I did not know about it, however it does not really work when ran within the context of an runOnJS execution, after a lot of no sense I figured out you can trigger their execution by running `jest.runAllTimers`, it seems to be only real way to test this functions.
